server.js

```diff
@@ -165,67 +157,106 @@ var CBrequestHandler = function(req, response, service, options, unit) {
     });
 }
 
-// Auxiliar function that handles generic requests (no ContextBroker)
-var requestHandler = function(req, response, options, unit) {
-    var userID = req.get('X-Actor-ID');
-    var API_KEY = req.get('X-API-KEY');
-    var publicPath = req.path;
+/**
+ * Auxiliar function that handles generic requests (no ContextBroker request).
+ * 
+ * @param  {Object} req     Incoming request.
+ * @param  {Object} res     Outgoing response
+ * @param  {Object} options Options for make the request to the endpoint.
+ * @param  {string} unit    Unit for the accounting.
+ */
+var requestHandler = function(req, res, options, unit) {
+    var user = req.get('X-Actor-ID');
+    var apiKey = req.get('X-API-KEY');
+    var body = req.body;
 
     request(options, function(error, resp, body) {
         if (error) {
-            response.status(500).send(error);
+            res.status(500).send(error);
```

Comment:
Cuando no puedes conectar con el servidor upstream no debes devolver un error 500 genérico. Puedes devolver un 504: Gateway Timeout - The server was acting as a gateway or proxy and did not receive a timely response from the upstream server.